### PR TITLE
Wrote tree traversal, updated readme, and worked on a test, the latter ofwhich still has errors

### PR DESCRIPTION
**Pull request by SharinaS**
- [x] Top-level README “Table of Contents” is updated
 - [ ] Feature tasks for this challenge are completed
 - [ ] Unit tests written and passing
     - [ ] “Happy Path” - Expected outcome
     - [ ] Expected failure
     - [ ] Edge Case (if applicable/obvious)
 - [x] README for this challenge is complete
     - [x] Summary, Description, Approach & Efficiency, Solution
     - [x] Link to code
     - [x] Picture of whiteboard

### CODE REVIEW

File: code401Challenges/src/main/java/code401Challenges/tree/Tree.java [hunk 1]
```diff
@@ -100,4 +99,26 @@ public Tree(Node<T> root) {
...
     }
 
+    // ======= Breadth First Traversal, using a queue, for numbers
+    public void breadthFirstTraverse (Node<Integer> root) {
+        Queue<Integer> storageQ = new Queue<>();
+
+        if (root == null) {
+            throw new NoSuchElementException();
+        } else {
+            storageQ.enqueue(root.value);
```
Comment (jeffborda) [inline code401Challenges/src/main/java/code401Challenges/tree/Tree.java]: We don't just want to enqueue the value, but the whole Node, because when we take it out, we will need to enqueue it's left and right children

File: code401Challenges/src/main/java/code401Challenges/tree/Tree.java [hunk 2]
```diff
@@ -100,4 +99,26 @@ public Tree(Node<T> root) {
...
+        }
+
+        while (!storageQ.isEmpty()) {
+            Integer currNode = storageQ.dequeue();
```
Comment (jeffborda) [inline code401Challenges/src/main/java/code401Challenges/tree/Tree.java]: Instead of type Integer, you can change it to: Node<T> curr = .....

File: code401Challenges/src/main/java/code401Challenges/tree/Tree.java [hunk 3]
```diff
@@ -100,4 +99,26 @@ public Tree(Node<T> root) {
...
+            System.out.println(currNode);
+            if (root.left != null) {
+                storageQ.enqueue(root.left.value);
```
Comment (jeffborda) [inline code401Challenges/src/main/java/code401Challenges/tree/Tree.java]: Then enqueue(currNode.left) and (currNode.right)

File: code401Challenges/src/main/java/code401Challenges/tree/Tree.java [hunk 4]
```diff
@@ -100,4 +99,26 @@ public Tree(Node<T> root) {
         return valuesArr;
     }
 
+    // ======= Breadth First Traversal, using a queue, for numbers
+    public void breadthFirstTraverse (Node<Integer> root) {
```
Comment (jeffborda) [inline code401Challenges/src/main/java/code401Challenges/tree/Tree.java]: You don't need to pass in a root node since you will be calling this method on a Tree, and all Trees have a root

File: code401Challenges/src/main/java/code401Challenges/tree/Tree.java (hunk 1 above)
Comment (SharinaS) [reply code401Challenges/src/main/java/code401Challenges/tree/Tree.java]: Oh! Good point. Thank you!
